lib/situation_room/site_controller.ex

```diff
@@ -0,0 +1,65 @@
+defmodule SituationRoom.Site.Controller do
+  @moduledoc """
+  Documentation for `SituationRoom.Site.Controller`.
+  """
+  use Plug.Router
+  alias SituationRoom.Site
+
+  plug(:match)
+
+  plug(Plug.Parsers,
+    parsers: [:urlencoded, :multipart],
+    pass: ["text/*"]
+  )
+
+  plug(:dispatch)
+
+  # Get all sites
+  get "/" do
+    send_resp(
+      conn,
+      200,
+      Jason.encode!(Site.get_all_sites())
+    )
+  end
+
+  # Get a site by id
+  get "/:id" do
+    case Site.get_site(id: conn.params["id"]) do
+      res = %SituationRoom.Site{} ->
+        send_resp(conn, 200, Jason.encode!(res))
+
+      nil ->
+        send_resp(conn, 404, "")
+    end
+  end
+
+  # This route can be hit when forward slashes are encoded as %2F
+  post "/" do
+    case Site.create_site(conn.params) do
+      {:ok, content} ->
+        send_resp(conn, 201, Jason.encode!(content))
+
+      _ ->
+        send_resp(conn, 400, "Cannot POST site")
```

Comment:
In the case of a create failure (on POST /sites) we need to let the user know why the create failed.  so this catchall needs to get the error  `{:error, changeset}`  and we probably want to return the `changeset.errors` in the JSON response I think